test/NuGet.Core.Tests/NuGet.Packaging.Test/NoRefOrLibFolderInPackageRuleTests.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+using NuGet.Packaging.Rules;
+using Moq;
+using System.Xml.Linq;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+
+namespace NuGet.Packaging.Test
+{
+    public class NoRefOrLibFolderInPackageRuleTests
+    {
+        [Theory]
+        [MemberData("WarningRaisedData", MemberType = typeof(FileSource))]
+        public void WarningRaisedWhenLibOrRefFolderWithTFMDataIsNotFoundAndBuildFolderIsFound(string[] files)
+        {
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        [Theory]
+        [MemberData("WarningNotRaisedData", MemberType = typeof(FileSource))]
+        public void WarningNotRaisedWhenLibOrRefFolderWithTFMDataIsFound(string[] files)
+        {
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        public static class FileSource
+        {
+            public static readonly List<object[]> WarningNotRaisedData
+                    = new List<object[]>
+                    {
+                    new object[] { new string[] {"build/random_tfm/test.props", "ref/random_tfm/test.dll"} },
+                    new object[] { new string[] {"build/random_tfm/test.props", "lib/random_tfm/test.dll"} },
+                    new object[] { new string[] {"ref/random_tfm/test.dll"} },
+                    new object[] { new string[] {"lib/random_tfm/test.dll"} },
+                    new object[] { new string[] {"build/any/test.props"} },
+                    new object[] { new string[] {"build/native/test.props"} }
+                    };
+
+            public static readonly List<object[]> WarningRaisedData
+                = new List<object[]>
+                {
+                new object[] { new string[] {"build/random_tfm/test.props"} },
```

Comment:
I just remembered that packages that use `build/` often have subdirectories like `build/bin/` or `build/lib/` that are used by their `build/packageid.targets` file, and since we're not warning on `build/packageid.targets`, we should not warn when the directory name under `build/` is not a valid TFM.

Therefore I don't believe that any of these `random_tfm` tests should warn since they're not real TFMs.

Reply:
and tests should be added for a package with `build/package.targets` and `build/bin/package.dll` that does not warn.

Reply:
So we would parse the framework, it would be a specific framework, but it would be "unsupported" so we can special unsupported. 

If it's a specific framework, and it's not "dotnet" or "native" or unsupported.